File: exer-pratica-especializacao/venda/Venda.java
package venda;

import livro.Livro;

public class Venda {
    
    private int quantidade_vendas;
    private Livro livro_vendido;
    private double valor_venda;

    /* Construtores */

    private void init(int quantidade_vendas, Livro livro_vendido){

        this.quantidade_vendas = quantidade_vendas;
        this.livro_vendido = livro_vendido;
        this.valor_venda = this.quantidade_vendas * livro_vendido.calculaPrecoLivro();
    }

    public Venda(int quantidade_vendas, Livro livro_vendido){
        init(quantidade_vendas, livro_vendido);
    }

    /* Getters e Setters */

    public Livro getLivro_vendido(){
        return this.livro_vendido;
    }

    public double getValor_venda(){
        return this.valor_venda;
    }

}

File: exer-pratica-especializacao/venda/venda.py
from livro.livro import Livro

class Venda:
    
    def __init__(self, quantidade_vendas : int, livro_vendido : Livro):
        
        valor_livro : float = livro_vendido.calculaPrecoLivro()
        
        self.quantidade_vendas = quantidade_vendas
        self.livro_vendido = livro_vendido
        self.valor_venda : float = self.quantidade_vendas * valor_livro
    
    # getters
    
    def getLivro_vendido(self):    
        return self.livro_vendido
    
    def getValor_venda(self):
        return self.valor_venda
    
    def getQuantidade_vendas(self):
        return self.quantidade_vendas